src/main/java/uk/gov/hmcts/reform/unspec/service/tasks/handler/StartBusinessProcessTaskHandler.java

```diff
@@ -19,33 +17,37 @@
 import uk.gov.hmcts.reform.unspec.service.CoreCaseDataService;
 import uk.gov.hmcts.reform.unspec.service.data.ExternalTaskInput;
 import uk.gov.hmcts.reform.unspec.service.flowstate.StateFlowEngine;
-import uk.gov.hmcts.reform.unspec.stateflow.StateFlow;
 
 import java.util.Map;
 
 @Component
 @RequiredArgsConstructor
-public class StartBusinessProcessTaskHandler implements ExternalTaskHandler {
+public class StartBusinessProcessTaskHandler implements BaseExternalTaskHandler {
 
     public static final String FLOW_STATE = "flowState";
     public static final String BUSINESS_PROCESS = "businessProcess";
     private final CoreCaseDataService coreCaseDataService;
     private final CaseDetailsConverter caseDetailsConverter;
-    private final ObjectMapper objectMapper;
+    private final ObjectMapper mapper;
     private final StateFlowEngine stateFlowEngine;
 
+    private VariableMap variables;
+
     @Override
-    public void execute(ExternalTask externalTask, ExternalTaskService externalTaskService) {
+    public void handleTask(ExternalTask externalTask) {
         CaseData caseData = startBusinessProcess(externalTask);
-        StateFlow stateFlow = stateFlowEngine.evaluate(caseData);
-        VariableMap variables = Variables.createVariables();
-        variables.putValue(FLOW_STATE, stateFlow.getState().getName());
-        externalTaskService.complete(externalTask, variables);
+        variables = Variables.createVariables();
+        variables.putValue(FLOW_STATE, stateFlowEngine.evaluate(caseData).getState().getName());
```

Comment:
Are we not better to move this into the parent class so it is made available to all our External Service Tasks once and for all?